### ISSUE TITLE
"version::CLASS" used only once when using Safe.pm

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#9829]
Migrated from [rt.perl.org#68538](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=68538) (status was 'resolved')

Searchable as RT68538$

**Comment by p5pRT** [comment]
### From eserte@biokovo-amd64.herceg.de

This is a bug report for perl from eserte@&#8203;biokovo\-amd64\.herceg\.de\,
generated with the help of perlbug 1\.39 running under perl 5\.10\.1\.




---
There's a new warning when just using Safe\.pm in some ways\. I
could reproduce it with the following piece of code

require Safe;
Safe\->new;
\_\_END\_\_

put into a file XXX\.pm and then using it like this&#8203;:

$ /usr/perl\-5\.10\.0\-3220\-gf9961a0/bin/perl \-w \-MXXX \-e1
Name "version&#8203;::CLASS" used only once&#8203;: possible typo at /usr/perl\-5\.10\.0\-3220\-gf9961a0/lib/5\.10\.1/amd64\-freebsd/Safe\.pm line 251\.

I did a bisect\, and then problem appeared in 1c92ff99e7dfbea7d6a8aebe7194d6251fb1b810
in blead\.

Regards\,
&nbsp;    Slaven



---


---
Flags&#8203;:
&nbsp;    category=library
&nbsp;    severity=low
&nbsp;    module=Safe


---
Site configuration information for perl 5\.10\.1&#8203;:

Configured by eserte at Tue Aug  4 10&#8203;:38&#8203;:36 CEST 2009\.

Summary of my perl5 \(revision 5 version 10 subversion 1\) configuration&#8203;:
&nbsp;  Commit id&#8203;: f9961a059c818ac71464be22671a6aeb834e3c87
&nbsp;  Platform&#8203;:
&nbsp;    osname=freebsd\, osvers=7\.0\-release\, archname=amd64\-freebsd
&nbsp;    uname='freebsd biokovo\-amd64\.herceg\.de 7\.0\-release freebsd 7\.0\-release \#0&#8203;: sun feb 24 10&#8203;:35&#8203;:36 utc 2008 root@&#8203;driscoll\.cse\.buffalo\.edu&#8203;:usrobjusrsrcsysgeneric amd64 '
&nbsp;    config\_args='\-Dprefix=/usr/perl\-5\.10\.0\-3220\-gf9961a0 \-Dusemymalloc=n \-D cc=ccache cc \-Dgccansipedantic \-de'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    useithreads=undef\, usemultiplicity=undef
&nbsp;    useperlio=define\, d\_sfio=undef\, uselargefiles=define\, usesocks=undef
&nbsp;    use64bitint=define\, use64bitall=define\, uselongdouble=undef
&nbsp;    usemymalloc=n\, bincompat5005=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='ccache cc'\, ccflags ='\-DHAS\_FPSETMASK \-DHAS\_FLOATINGPOINT\_H \-fno\-strict\-aliasing \-pipe \-fstack\-protector \-I/usr/local/include'\,
&nbsp;    optimize='\-O'\,
&nbsp;    cppflags='\-DHAS\_FPSETMASK \-DHAS\_FLOATINGPOINT\_H \-fno\-strict\-aliasing \-pipe \-fstack\-protector \-I/usr/local/include'
&nbsp;    ccversion=''\, gccversion='4\.2\.1 20070719  \[FreeBSD\]'\, gccosandvers=''
&nbsp;    intsize=4\, longsize=8\, ptrsize=8\, doublesize=8\, byteorder=12345678
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=16
&nbsp;    ivtype='long'\, ivsize=8\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=8
&nbsp;    alignbytes=8\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='ccache cc'\, ldflags ='\-Wl\,\-E  \-fstack\-protector \-L/usr/local/lib'
&nbsp;    libpth=/usr/lib /usr/local/lib
&nbsp;    libs=\-lgdbm \-lm \-lcrypt \-lutil \-lc
&nbsp;    perllibs=\-lm \-lcrypt \-lutil \-lc
&nbsp;    libc=\, so=so\, useshrplib=false\, libperl=libperl\.a
&nbsp;    gnulibc\_version=''
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags=' '
&nbsp;    cccdlflags='\-DPIC \-fPIC'\, lddlflags='\-shared  \-L/usr/local/lib \-fstack\-protector'

Locally applied patches&#8203;:
&nbsp;    PERL\_GIT\_UNPUSHED\_COMMITS        /\* do not remove this line \*/
&nbsp;    PERL\_GIT\_UNCOMMITTED\_CHANGES    /\* do not remove this line \*/



---
@&#8203;INC for perl 5\.10\.1&#8203;:
&nbsp;    /usr/perl\-5\.10\.0\-3220\-gf9961a0/lib/5\.10\.1/amd64\-freebsd
&nbsp;    /usr/perl\-5\.10\.0\-3220\-gf9961a0/lib/5\.10\.1
&nbsp;    /usr/perl\-5\.10\.0\-3220\-gf9961a0/lib/site\_perl/5\.10\.1/amd64\-freebsd
&nbsp;    /usr/perl\-5\.10\.0\-3220\-gf9961a0/lib/site\_perl/5\.10\.1
&nbsp;    \.



---
Environment for perl 5\.10\.1&#8203;:
&nbsp;    HOME=/home/e/eserte
&nbsp;    LANG \(unset\)
&nbsp;    LANGUAGE \(unset\)
&nbsp;    LC\_ALL=de\_DE\.ISO8859\-1
&nbsp;    LC\_CTYPE=de\_DE\.ISO8859\-1
&nbsp;    LD\_LIBRARY\_PATH \(unset\)
&nbsp;    LOGDIR \(unset\)
&nbsp;    PATH=/usr/local/bin&#8203;:/usr/bin&#8203;:/bin&#8203;:/usr/gnu/bin&#8203;:/usr/TeX/bin&#8203;:/usr/local/sbin&#8203;:/usr/sbin&#8203;:/sbin&#8203;:/usr/local/pilot/bin&#8203;:/home/e/eserte/bin/FreeBSD&#8203;:/home/e/eserte/bin/sh&#8203;:/home/e/eserte/bin&#8203;:/usr/X11R6/bin&#8203;:/usr/X11/bin&#8203;:/usr/X386/bin&#8203;:/usr/games&#8203;:/home/e/eserte/devel
&nbsp;    PERLDOC=\-MPod&#8203;::Perldoc&#8203;::ToTextOverstrike
&nbsp;    PERL\_BADLANG \(unset\)
&nbsp;    PERL\_HTML\_DISPLAY\_CLASS=HTML&#8203;::Display&#8203;::Mozilla
&nbsp;    SHELL=/bin/tcsh

**Comment by p5pRT** [comment]
### From @iabyn

On Fri\, Aug 14\, 2009 at 11&#8203;:50&#8203;:43PM \-0700\, eserte@&#8203;biokovo\-amd64\.herceg\.de \(via RT\) wrote&#8203;:
> \# New Ticket Created by  eserte@&#8203;biokovo\-amd64\.herceg\.de 
> \# Please include the string&#8203;:  \[perl \#68538\]
> \# in the subject line of all future correspondence about this issue\. 
> \# \<URL&#8203;: http&#8203;://rt\.perl\.org/rt3/Ticket/Display\.html?id=68538 >
> 
> 
> This is a bug report for perl from eserte@&#8203;biokovo\-amd64\.herceg\.de\,
> generated with the help of perlbug 1\.39 running under perl 5\.10\.1\.
> 
> 
> \-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
> There's a new warning when just using Safe\.pm in some ways\. I
> could reproduce it with the following piece of code
> 
> require Safe;
> Safe\->new;
> \_\_END\_\_
> 
> put into a file XXX\.pm and then using it like this&#8203;:
> 
> $ /usr/perl\-5\.10\.0\-3220\-gf9961a0/bin/perl \-w \-MXXX \-e1
> Name "version&#8203;::CLASS" used only once&#8203;: possible typo at /usr/perl\-5\.10\.0\-3220\-gf9961a0/lib/5\.10\.1/amd64\-freebsd/Safe\.pm line 251\.
> 
> I did a bisect\, and then problem appeared in 1c92ff99e7dfbea7d6a8aebe7194d6251fb1b810
> in blead\.

Now fixed in blead with a930c511fcc6cf6f5ef849a9a4a028ff1cd6e27a
and fixed in 5\.10\.1\-to\-be with 9de9eeb1452212b331264652386099df5794d4a5\.


\-\- 
All wight\. I will give you one more chance\. This time\, I want to hear
no Wubens\. No Weginalds\. No Wudolf the wed\-nosed weindeers\.
&nbsp;    \-\- Life of Brian

**Comment by p5pRT** [comment]
The RT System itself - Status changed from 'new' to 'open'

**Comment by p5pRT** [comment]
### From @tonycoz

On Sun Aug 16 16&#8203;:54&#8203;:40 2009\, davem wrote&#8203;:
> Now fixed in blead with a930c511fcc6cf6f5ef849a9a4a028ff1cd6e27a
> and fixed in 5\.10\.1\-to\-be with
> 9de9eeb1452212b331264652386099df5794d4a5\.

Tests ok in current blead \(pre\-5\.18\)\, so closing\.

Tony

**Comment by p5pRT** [comment]
@tonycoz - Status changed from 'open' to 'resolved'
